### ISSUE TITLE
Problem with using model.history.append()

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#2136]
JIRA ticket JP-325 (https://jira.stsci.edu/browse/JP-325) reports:
```
In older versions of the pipeline, I was able to append strings to the history of datamodels,
for example:

In [1]: from jwst.datamodels import GainModel
In [2]: mod = GainModel()
In [3]: mod.history.append('Add this to the history')
But I am getting the following error now:

In [1]: from jwst.datamodels import GainModel
In [2]: mod = GainModel()
In [3]: mod.history.append('Testing')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-b66a027220ee> in <module>()
----> 1 mod.history.append('Testing')

AttributeError: 'dict' object has no attribute 'append'

Did the datamodels change, or is there an issue with my environment? I know you can use 
the add_history attribute of FITS files to do the same thing (e.g., hdulist = fits.open('file.fits'),
hdulist[0].header.add_history('Add this to the history')), but several of our scripts append it 
using the datamodel attribute and I just want to check before I make any changes.
```

**Comment by bernie-simon** [comment]
Datamodels is like the castle in the Gormenghast trilogy and a change over here can have an unexpected result over there. I'll be looking into the reason.

**Comment by nden** [comment]
`AsdfFile.add_history_entry` has changed format from `list` to `dict`. `DataModel.history` was changed in #1879 [here](https://github.com/STScI-JWST/jwst/commit/81f9fbd455ddbcf5500a0728311a1aeb2edd6964#diff-458aa15d1359866095ef3864c45d9f45) . 
I see now that there was another change to the asdf format in spacetelescope/asdf#501 . FYI, in case this helps.

**Comment by nden** [comment]
The current asdf history format is `{'entries': [], 'extensions': []}`. `DataModel.history['entries']` is not created by deault. One possible solution is to create a default `DataModel.history['entries'] = []` in the data model with users doing `history['entries'].append('More History')`.

**Comment by drdavella** [comment]
👍 for the Gormenghast reference.

**Comment by nden** [comment]
#fixed in #2142
